spirv_msl.cpp

```diff
@@ -3793,10 +3807,30 @@ string CompilerMSL::entry_point_args(bool append_comma)
 			auto &m = meta.at(type.self);
 			if (m.members.size() == 0)
 				break;
-			if (!ep_args.empty())
-				ep_args += ", ";
-			ep_args += get_argument_address_space(var) + " " + type_to_glsl(type) + "& " + r.name;
-			ep_args += " [[buffer(" + convert_to_string(r.index) + ")]]";
+			if (!type.array.empty())
+			{
+				// Metal doesn't directly support this, so we must expand the
+				// array. We'll declare a local array to hold these elements
+				// later.
+				uint32_t array_size =
```

Comment:
Unsized array (bindless) or array-of-array won't work here. You should throw error for that case.